core/src/main/java/com/radixdlt/consensus/liveness/MultiFactorPacemakerTimeoutCalculator.java

```diff
@@ -64,58 +64,53 @@
- * round.
+ * Main timeout calculator implementation: uses the number of consecutive timeout occurrences to
+ * calculate an exponential timeout and then on top of that applies a multiplier that's based on
+ * vertex store utilization.
  */
-public final class ExponentialPacemakerTimeoutCalculator implements PacemakerTimeoutCalculator {
-  private final long baseTimeoutMilliseconds;
-  private final double rate;
-  private final int maxExponent;
-  private final long additionalRoundTimeIfProposalReceivedMs;
+public final class MultiFactorPacemakerTimeoutCalculator implements PacemakerTimeoutCalculator {
+  private final PacemakerTimeoutCalculatorConfig config;
 
   @Inject
-  public ExponentialPacemakerTimeoutCalculator(
-      @PacemakerBaseTimeoutMs long baseTimeoutMilliseconds,
-      @PacemakerBackoffRate double rate,
-      @PacemakerMaxExponent int maxExponent,
-      @AdditionalRoundTimeIfProposalReceivedMs long additionalRoundTimeIfProposalReceivedMs) {
-    if (baseTimeoutMilliseconds <= 0) {
-      throw new IllegalArgumentException(
-          "timeoutMilliseconds must be > 0 but was " + baseTimeoutMilliseconds);
-    }
-    if (rate <= 1.0) {
-      throw new IllegalArgumentException("rate must be > 1.0, but was " + rate);
-    }
-    if (maxExponent < 0) {
-      throw new IllegalArgumentException("maxExponent must be >= 0, but was " + maxExponent);
-    }
-    double maxTimeout = baseTimeoutMilliseconds * Math.pow(rate, maxExponent);
-    if (maxTimeout > Long.MAX_VALUE) {
-      throw new IllegalArgumentException(
-          "Maximum timeout value of " + maxTimeout + " is too large");
-    }
-
-    this.baseTimeoutMilliseconds = baseTimeoutMilliseconds;
-    this.rate = rate;
-    this.maxExponent = maxExponent;
-    this.additionalRoundTimeIfProposalReceivedMs = additionalRoundTimeIfProposalReceivedMs;
+  public MultiFactorPacemakerTimeoutCalculator(PacemakerTimeoutCalculatorConfig config) {
+    this.config = config;
   }
 
   @Override
-  public long calculateTimeoutMs(long timeoutOccurrences) {
-    double exponential = Math.pow(this.rate, Math.min(this.maxExponent, timeoutOccurrences));
-    return Math.round(this.baseTimeoutMilliseconds * exponential);
+  @SuppressWarnings("UnstableApiUsage")
+  public long calculateTimeoutMs(long timeoutOccurrences, double vertexStoreUtilizationRatio) {
+    final var exponential =
+        Math.pow(config.rate(), Math.min(config.maxExponent(), timeoutOccurrences));
+
+    final var exponentialTimeout = config.baseTimeoutMs() * exponential;
+
+    // It should already be in the [0, 1] range, but we're nonetheless sanitizing the input
+    final var vertexStoreUtilizationRatioClamped =
+        Doubles.constrainToRange(vertexStoreUtilizationRatio, 0, 1);
+
+    // We're only applying the multiplier if vertexStoreUtilizationRatio is
+    // on or above vertexStoreMultiplierThreshold: we're translating from
+    // range [vertexStoreMultiplierThreshold, 1] to [1, maxVertexStoreMultiplier].
```

Comment:
I was partially expecting to see an exponential here 

(So roughly take linearly map: `[vertexStoreMultiplierThreshold, 1]` to `exponent: [0, maxVertexStoreExponent]` and then return `Math.round(config.baseTimeoutMs() * timeoutExponential * vertexSizeExponential);`)

But I think linear is fine too and does its job.